### PR TITLE
[CST-1068] Enable admins to transfer participants

### PR DESCRIPTION
**Pull request by tonyheadford**
### Context

- Ticket: [Jira](https://dfedigital.atlassian.net/browse/CST-1068)

### Changes proposed in this pull request
Add functionality to admin console to transfer a participant to a different school

### Guidance to review

image
image
image
image
image
image
image

### CODE REVIEW

File: app/services/identity/create.rb [hunk 1]
```diff
@@ -20,7 +20,10 @@ def initialize(user:, origin: :ecf, email: nil)
       @email = email || user.email
     end
 
-    def find_or_create_identity
+    def find_or_create_identity!
+      email_user = Identity.find_user_by(email:)
+      raise "Email already taken" if email_user.present? && email_user != user
```
Comment (peteryates) [inline app/services/identity/create.rb:25]: Not one for now but we do this in a few places, could be worth creating a service class

Comment (tonyheadford) [reply app/services/identity/create.rb:25]: Yes good idea, I noticed this seemed to be missing so shoved it in with this PR 👍

File: app/forms/school_transfer_form.rb [hunk 2]
```diff
@@ -0,0 +1,189 @@
+# frozen_string_literal: true
+
+class SchoolTransferForm
```
[the remark this answers is not in the document]
Comment (tonyheadford) [reply app/forms/school_transfer_form.rb:3]: Yes good idea 👍

File: app/services/induction/transfer_and_continue_existing_programme.rb [hunk 3]
```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+
+class Induction::TransferAndContinueExistingProgramme < BaseService
```
Comment (ltello) [inline app/services/induction/transfer_and_continue_existing_programme.rb:3]: This service class is really similar to a couple of them for schools to transfer participants. Cannot we merge the 3 in just one or maybe 2?

File: app/forms/school_transfer_form.rb [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+
+  def previous_step
+    return if current_step == STEPS.first
+
+    step = nil
+    STEPS.each_with_index do |step_name, i|
+      step = STEPS[i - 1] and break if step_name == current_step
+    end
+    step
+  end
```
[the remark this answers is not in the document]
Comment (tonyheadford) [reply app/forms/school_transfer_form.rb]: Yes that's much nicer. Remind me to show you how to use the suggestion functionality in github as I could've just committed your suggestion 👍

**Review by ltello — APPROVED** [review]

**Pull request merged**